### PR TITLE
Login Logout changes for new approve org. webapp.

### PR DESCRIPTION
**Pull request by rtiru-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
[EUI-5862](https://tools.hmcts.net/jira/browse/EUI-5862)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### Console log ###
```
Rohiths-Air:rpx-xui-approve-org rohithtiru$ HMCTS_EMAIL=rohith.tiru1@hmcts.net HMCTS_PASSWORD=****** TEST_URL=https://administer-orgs.demo.platform.hmcts.net/  yarn run test:smoke
yarn run v1.22.18
warning package.json: "dependencies" has dependency "mochawesome" with range "^4.1.0" that collides with a dependency in "devDependencies" of the same name with version "3.1.1"
warning ../../../../package.json: No license field
$ webdriver-manager update --versions.chrome 2.40 && protractor ./test/e2e/config/smoke.conf.js
[11:35:35] I/update - chromedriver: file exists /Users/rohithtiru/workspace/solirius/repos/rpx-xui-approve-org/node_modules/webdriver-manager/selenium/chromedriver_2.40.zip
[11:35:35] I/update - chromedriver: unzipping chromedriver_2.40.zip
[11:35:36] I/update - chromedriver: setting permissions to 0755 for /Users/rohithtiru/workspace/solirius/repos/rpx-xui-approve-org/node_modules/webdriver-manager/selenium/chromedriver_2.40
[11:35:36] I/update - chromedriver: chromedriver_2.40 up to date
[11:35:36] I/update - selenium standalone: file exists /Users/rohithtiru/workspace/solirius/repos/rpx-xui-approve-org/node_modules/webdriver-manager/selenium/selenium-server-standalone-3.141.59.jar
[11:35:36] I/update - selenium standalone: selenium-server-standalone-3.141.59.jar up to date
[11:35:37] I/update - geckodriver: file exists /Users/rohithtiru/workspace/solirius/repos/rpx-xui-approve-org/node_modules/webdriver-manager/selenium/geckodriver-v0.31.0.tar.gz
[11:35:37] I/update - geckodriver: unzipping geckodriver-v0.31.0.tar.gz
[11:35:37] I/update - geckodriver: setting permissions to 0755 for /Users/rohithtiru/workspace/solirius/repos/rpx-xui-approve-org/node_modules/webdriver-manager/selenium/geckodriver-v0.31.0
[11:35:37] I/update - geckodriver: geckodriver-v0.31.0 up to date
[11:35:40] I/launcher - Running 1 instances of WebDriver
[11:35:40] I/direct - Using ChromeDriver directly...
(node:29674) DeprecationWarning: cucumber: defineSupportCode is deprecated. Please require/import the individual methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Feature: Login

  @fullfunctional @login
  Scenario: un-authenticated user login
    When I navigate to EUI Approve Organisation Url
11:35:50 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, #i0116)
11:35:50 GMT+0100 (British Summer Time) : wait done for sec50
11:35:51 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, #i0118)
11:35:51 GMT+0100 (British Summer Time) : wait done for sec50
11:35:52 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, #idSIButton9)
11:35:53 GMT+0100 (British Summer Time) : wait done for sec50
11:35:53 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:00 GMT+0100 (British Summer Time) : wait done for sec50
11:36:00 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:00 GMT+0100 (British Summer Time) : wait done for sec50
    Then I am on Idam login page
    When I enter an Invalid email-address and password to login
    Then I should be redirected to the Idam login page
11:36:00 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, h1.heading-large)
11:36:01 GMT+0100 (British Summer Time) : wait done for sec50
    Then I should see failure error summary

  @login
  Scenario: login and log out from approve organisation as HMCTS Admin user
    When I navigate to EUI Approve Organisation Url
11:36:03 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:04 GMT+0100 (British Summer Time) : wait done for sec50
11:36:04 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:04 GMT+0100 (British Summer Time) : wait done for sec50
    Given I am logged into approve organisation with HMCTS admin
11:36:04 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:04 GMT+0100 (British Summer Time) : wait done for sec50
TimeoutError: Element still present : By(css selector, input#username)
Wait timed out after 5000ms email field is still present with empty value indicating  Login page reloaded due to EUI-1856 : Login re attempt 1
11:36:16 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:16 GMT+0100 (British Summer Time) : wait done for sec50
ONE ATTEMPT:  EUI-1856 issue occured / total logins => 1 / 1
TWO ATTEMPT: EUI-1856 issue occured / total logins => 0 / 1
    Then I should be redirected to approve organisation dashboard page
11:36:20 GMT+0100 (British Summer Time) : Started step
11:36:26 GMT+0100 (British Summer Time) : browser readyState value  "complete"
11:36:26 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(xpath, //a[@class='hmcts-header__link'])
11:36:26 GMT+0100 (British Summer Time) : wait done for sec50
    When I select the sign out link
    Then I should be redirected to the Idam login page
11:36:40 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, h1.heading-large)
11:36:40 GMT+0100 (British Summer Time) : wait done for sec50

  @login
  Scenario: login and log out from approve organisation as FR user
    When I navigate to EUI Approve Organisation Url
11:36:41 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:42 GMT+0100 (British Summer Time) : wait done for sec50
11:36:42 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:42 GMT+0100 (British Summer Time) : wait done for sec50
    Given I am logged into approve organisation with FR judge details
11:36:42 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:36:42 GMT+0100 (British Summer Time) : wait done for sec50
ONE ATTEMPT:  EUI-1856 issue occured / total logins => 1 / 2
TWO ATTEMPT: EUI-1856 issue occured / total logins => 0 / 2
    Then I should be redirected to approve organisation dashboard page
11:36:44 GMT+0100 (British Summer Time) : Started step
11:36:50 GMT+0100 (British Summer Time) : browser readyState value  "complete"
11:36:50 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(xpath, //a[@class='hmcts-header__link'])
11:36:50 GMT+0100 (British Summer Time) : wait done for sec50
    When I select the sign out link
    Then I should be redirected to the Idam login page
11:37:04 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, h1.heading-large)
11:37:04 GMT+0100 (British Summer Time) : wait done for sec50

  @fullfunctional @login
  Scenario: Verify the direct link navigate to login page
    When I navigate to EUI Approve Organisation Url
11:37:05 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:37:06 GMT+0100 (British Summer Time) : wait done for sec50
11:37:06 GMT+0100 (British Summer Time) : Before starting wait for element 50 : By(css selector, input#username)
11:37:06 GMT+0100 (British Summer Time) : wait done for sec50
    Given I navigate to approve organisation Url direct link
    Then I should be redirected back to Login page after direct link

4 scenarios (4 passed)
18 steps (18 passed)
1m44.472s


=====================================================================================
    Multiple Cucumber HTML report generated in:

    /Users/rohithtiru/workspace/solirius/repos/rpx-xui-approve-org/reports/tests/smoke/index.html
=====================================================================================
```

### CODE REVIEW

File: test/e2e/features/step_definitions/steps/approveOrganisation.steps.js [hunk 1]
```diff
@@ -20,10 +20,15 @@ defineSupportCode(function ({ Given, When, Then }) {
 
     When(/^I navigate to EUI Approve Organisation Url$/, { timeout: 600 * 1000 }, async function () {
         await browser.get(config.config.baseUrl) ;
-        await browser.driver.manage()
+        if (!process.env.TEST_URL.includes('demo')){ //Do not delete cookies for demo env.
```
Comment (sreekanthpuligadda) [inline test/e2e/features/step_definitions/steps/approveOrganisation.steps.js:23]: If cookies are not cleared before starting a scenario, in current scenario we need to check if there is a login step coming up next does not fail scenario. Are the affects of this change addressed?

Comment (rtiru-hmcts) [reply test/e2e/features/step_definitions/steps/approveOrganisation.steps.js:23]: @sreekanthpuligadda This was verified prior to making the change. For approve-org, I commented out logout for one of  the scenario and tested. The next scenario it executed was successful. The cookie persistence is only for demo env (inside if condition). I believe this is a known issue with demo env.  Is there any other test i would need to run and check?

**Review by Vamshi-Muniganti — APPROVED** [review]
looks good to me

File: test/e2e/features/step_definitions/steps/approveOrganisation.steps.js (hunk 1 above)
Comment (Vamshi-Muniganti) [reply test/e2e/features/step_definitions/steps/approveOrganisation.steps.js:23]: this is just workaround/additional code anyways it wont break any FT's on master/nightly/PR pipelines and demo doesn't have any pipeline itself, it helps when we run tests local pointing to demo and Rohith ran all test suite and all passed, there is no scenario without a login page if cookie is cleared,

**Review by sreekanthpuligadda — APPROVED** [review]
LGTM

**Review by Vamshi-Muniganti — APPROVED** [review]
looks good

**Review by sreekanthpuligadda — APPROVED** [review]
LGTM

**Pull request merged**